src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -132,7 +145,24 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
             try
             {
                 var originalMethod = context.Request.Method;
-                await next(context).ConfigureAwait(false);
+                if (_options.CaptureBlockingCalls && _monitor is not null)
+                {
+                    var syncCtx = SynchronizationContext.Current;
+                    SynchronizationContext.SetSynchronizationContext(syncCtx == null ? _detectBlockingSyncCtx : new DetectBlockingSynchronizationContext(_monitor, syncCtx));
```

Comment:
@bruno-garcia won't the SynchronizationContext in ASP.NET Core always be null? Is this code here just in case SDK users have implemented a custom sync context for some reason?